Text/HealthcareTA/HealthcareTA.cs

```diff
@@ -0,0 +1,136 @@
+    {
+        private static readonly string healthcareApiEnvEndpoint = "HEALTHCARE_API_ENDPOINT";
+        private static readonly string healthcareApiEnvKey = "HEALTHCARE_API_KEY";
+        private static readonly int defaultTimeout = 230;
+        private static readonly int maxCharLength = 5000;
+
+        [FunctionName("HealthcareTA")]
+        public static async Task<IActionResult> Run(
+            [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequest req,
+            ILogger log,
+            ExecutionContext executionContext)
+        {
+            string skillName = executionContext.FunctionName;
+            IEnumerable<WebApiRequestRecord> requestRecords = WebApiSkillHelpers.GetRequestRecords(req);
+            if (requestRecords == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Invalid request record array.");
+            }
+
+            // Get Endpoint and access key from App Settings
+            string apiKey = Environment.GetEnvironmentVariable(healthcareApiEnvKey, EnvironmentVariableTarget.Process);
+            string apiEndpoint = Environment.GetEnvironmentVariable(healthcareApiEnvEndpoint, EnvironmentVariableTarget.Process);
+            if (apiKey == null || apiEndpoint == null)
+            {
+                return new BadRequestObjectResult($"{skillName} - Healthcare Text Analytics API key or endpoint is missing. Make sure to set them in the Environment Variables.");
+            }
+
+            var client = new TextAnalyticsClient(new Uri(apiEndpoint), new AzureKeyCredential(apiKey));
+
+            // Get a custom timeout from the header, if it exists. If not use the default timeout.
+            int timeout;
+            if (!int.TryParse(req.Headers["SkillTimeout"].ToString(), out timeout))
+            {
+                timeout = defaultTimeout;
+            }
+            var timeoutMiliseconds = timeout * 1000;
+
+            WebApiSkillResponse response = await WebApiSkillHelpers.ProcessRequestRecordsAsync(skillName, requestRecords,
+                async (inRecord, outRecord) => {
+                    // Prepare analysis operation input
+                    var document = inRecord.Data["document"] as string;
+                    var options = new AnalyzeHealthcareEntitiesOptions { };
+                    if (document.Length >= maxCharLength)
+                    {
+                        outRecord.Warnings.Add(new WebApiErrorWarningContract
+                        {
+                            Message = $"Healthcare Text Analytics Error: The submitted document was over {maxCharLength} characters. It has been truncated to fit this requirement."
+                        });
+                        document = document.Substring(0, maxCharLength);
+                    }
+                    List<string> batchInput = new List<string>()
+                    {
+                        document
+                    };
```

Comment:
I do think we should at the very least make a follow up work item to see if we can batch multiple documents into a single request here

Reply:
Created a work item for this: https://msdata.visualstudio.com/Azure%20Search/_workitems/edit/1251217